[TS-886] Undefined fields check option (#8)
th2-net/th2-codec-open-api: README.md

All messages processing by encode must contain those types.

> Encode accepts only parsed messages, all raw will be passed.

Reviewer: ```suggestion
Encode accepts only parsed messages, raw ones will be skipped.
```

### Codec configs:

> * warnUndefinedFields - Enable or Disable warnings for all undefined fields inside object structures, true by default.

Reviewer: ```suggestion
* checkUndefinedFields - Enable or Disable warnings for all undefined fields inside object structures, true by default.
```

### Configuration example

```yaml
checkUndefinedFields: true
validationSettings:
  enableRecommendations: true
dictionaryParseOption:
  resolve: true
```

May be empty due to missing required fields

## Release notes

### 0.2.0

> + Feature: check undefined fields and throws errors if they are found

Reviewer: ```suggestion
+ Feature: check for undefined fields and throw errors if they are found
```
